src/lib/dialog/dialog-ref.ts

```diff
@@ -101,14 +101,14 @@ export class MatDialogRef<T> {
   /**
    * Gets an observable that is notified when the dialog is finished closing.
    */
-  afterClosed(): Observable<any> {
+  afterClosed(): Observable<R> {
```

Comment:
Should the return type here be `Observable<R | undefined>` since the `dialogResult` parameter in the `close` method is optional?

Reply:
Good point, I'm not sure why TS didn't complain about it.